apps/costcontrol/test/unit/startup_test.js

```diff
@@ -280,7 +278,7 @@ suite('Application Startup Modes Test Suite >', function() {
         assertFTEStarted(applicationMode, done);
 
         // Launch the second start-up
-        MockNavigatorMozIccManager.triggerEventListeners('iccdetected', {});
+        window.dispatchEvent(new Event('dataAirplaneModeDisabled'));
```

Comment:
Let's rethink this test a little bit. First and most important, move the assert at the end of the function to reinforce it is the last thing that should happen. Now, notice [line 256](https://github.com/mozilla-b2g/gaia/pull/27046/files#diff-84507f20c160644872a74f6ae2231da3L256) and [line 260](https://github.com/mozilla-b2g/gaia/pull/27046/files#diff-84507f20c160644872a74f6ae2231da3L260). Forcing airplane mode to be enabled is one case of failure and making the ICC request to fail is another different. Indeed, due to the new conditional in [line 484](https://github.com/mozilla-b2g/gaia/pull/27046/files#diff-badba8f26bdcdabb07c28ca9514c9d60R484), enabling airplane mode avoid completely to call the `requestDataSimIcc` function.

So, let's split this test case in two: one keeping the failing request to `requestDataSimIcc` and another dedicated for the airplane mode. What do you think?